src/lib.rs

```diff
@@ -1,40 +1,97 @@
-extern crate byteorder;
-#[macro_use]
-extern crate enum_primitive;
-extern crate num;
+use std::any::type_name;
+use std::io::{Read, Seek};
 
-use std::fmt;
-use std::io::Result;
+use num_traits::FromPrimitive;
+use tiff::decoder::{Decoder, DecodingResult};
+use tiff::tags::Tag;
+use tiff::TiffResult;
 
-pub mod geotiff;
-mod lowlevel;
-mod reader;
+use crate::image_data::*;
 
-pub use geotiff::TIFF;
-use reader::*;
+mod image_data;
 
-/// The GeoTIFF library reads `.tiff` files.
+macro_rules! assert_type {
+    ($result: expr, $actual: ty, $expected: ty) => {
+        $result
+            .ok_or_else(|| {
+                format!(
+                    "Cannot represent {} as {}",
+                    type_name::<$actual>(),
+                    type_name::<$expected>()
+                )
+            })
+            .unwrap()
```

Comment:
The previous `get_value_at` method was returning a value directly, so using `.unwrap()` here makes sense to match the same behaviour. But maybe it would be better to return a `TiffResult` once the panic at https://github.com/georust/geotiff/pull/17/files#r1746172003 is removed.

Reply:
As mentioned above, I'd use a separate `GeoTiffError` type for errors specific to this crate.

Reply:
Ok with leaving this `.unwrap()` for now as mentioned at https://github.com/georust/geotiff/pull/17#discussion_r1746581439.